components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -0,0 +1,304 @@
+     * This method will validate the following conditions.
+     *  - Whether the client id sent in the request and client id retrieved from the database are equal
+     *  - Whether the difference between two dates is less than the configured time
+     *  - Whether payloads are equal
+     *
+     * @param consentManageData        Consent Manage Data
+     * @param consentResource          Detailed Consent Resource
+     * @return  IdempotencyValidationResult
+     */
+    private IdempotencyValidationResult validateIdempotencyConditions(ConsentManageData consentManageData,
+                                                                      DetailedConsentResource consentResource)
+            throws IdempotencyValidationException, IOException {
+        // Compare the client ID sent in the request and client id retrieved from the database
+        // to validate whether the request is received from the same client
+        if (IdempotencyValidationUtils.isClientIDEqual(consentResource.getClientID(),
+                consentManageData.getClientId())) {
+            // Check whether difference between two dates is less than the configured time
+            if (IdempotencyValidationUtils.isRequestReceivedWithinAllowedTime(getCreatedTimeOfPreviousRequest(
+                    consentManageData.getRequestPath(), consentResource.getConsentID()))) {
+                // Compare whether JSON payloads are equal
+                if (isPayloadSimilar(consentManageData, getPayloadOfPreviousRequest(
+                        consentManageData.getRequestPath(), consentResource.getConsentID()))) {
+                    log.debug("Payloads are similar and request received within allowed" +
+                            " time. Hence this is a valid idempotent request");
+                    return new IdempotencyValidationResult(true, true,
+                            consentResource, consentResource.getConsentID());
+                } else {
+                    log.error(IdempotencyConstants.ERROR_PAYLOAD_NOT_SIMILAR);
+                    throw new IdempotencyValidationException(IdempotencyConstants
+                            .ERROR_PAYLOAD_NOT_SIMILAR);
+                }
+            } else {
+                log.error(IdempotencyConstants.ERROR_AFTER_ALLOWED_TIME);
+                throw new IdempotencyValidationException(IdempotencyConstants
+                        .ERROR_AFTER_ALLOWED_TIME);
+            }
+        } else {
+            log.error(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
+            throw new IdempotencyValidationException(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
+        }
+    }
+
+    /**
+     * Method to get the Idempotency Attribute Name store in consent Attributes.
+     *
+     * @param resourcePath     Resource Path
+     * @return idempotency Attribute Name.
+     */
+    public String getIdempotencyAttributeName(String resourcePath) {
```

Comment:
The argument is not used in the method.

Reply:
This is the sample implementation for Accelerator. For Toolkits, it may require the resource path. Hence, passing is as an argument

Reply:
Shall we make this method protected?

Reply:
FIxed https://github.com/wso2-extensions/bfsi-consent-management/pull/10/commits/6b273a8368789eaf763b1691adfe30a06f507ff3